### PR TITLE
MEN-4562: Remove matching update_control_maps for finished deployments

### PR DESCRIPTION
**Pull request by lluiscampos**
And add simple unit test for updatemanager.Delete

### CODE REVIEW

**Comment by lluiscampos** [comment]
Acceptance tests in https://github.com/mendersoftware/meta-mender/pull/1385

File: app/updatemanager.go [hunk 1]
```diff
@@ -459,6 +459,27 @@ func (c *ControlMapPool) ClearExpired() {
 	c.saveToStore()
 }
 
+// Removes maps from the pool for which ID match deploymentId
+func (c *ControlMapPool) ClearIdMatchingDeploymentId(deploymentId string) {
```
Comment (oleorhagen) [inline app/updatemanager.go:463]: How is this different from this: https://github.com/mendersoftware/mender/blob/d2c66f51a2e336337f0ee9daf133eeb3dba7b9ab/app/updatemanager.go#L421

:thinking:

Comment (lluiscampos) [reply app/updatemanager.go:463]: In that mine is documented? :laughing: 

Nah, I wrote this long ago (predates `Delete`) and I didn't realize while rebasing. Removing it now...

Comment (lluiscampos) [reply app/updatemanager.go:463]: Not only mine is documented, but unit tested! Which I see is not the case for `Delete`...

**Review by oleorhagen — APPROVED** [review]

**Review by kacf — APPROVED** [review]
Strictly speaking I'm not sure we need a changelog for this, since this is part of a whole new feature.

**Comment by lluiscampos** [comment]
> Strictly speaking I'm not sure we need a changelog for this, since this is part of a whole new feature.

Right, I agree. Removing...

**Comment by lluiscampos** [comment]
Testing together with https://github.com/mendersoftware/meta-mender/pull/1392 @ [Pipeline 318490283](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/318490283)

**Comment by lluiscampos** [comment]
Rebased on top of #800

**Comment by lluiscampos** [comment]
New build at [Pipeline 319352823](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/319352823)

**Comment by lluiscampos** [comment]
New build at [Pipeline 319815091](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/319815091)

**Pull request merged**